src/dupin.cpp

```diff
@@ -0,0 +1,176 @@
+#include "dupin.h"
+#include <Eigen/Dense>
+#include <iomanip>
+#include <iostream>
+#include <limits>
+#include <tbb/blocked_range2d.h>
+#include <tbb/global_control.h>
+#include <tbb/parallel_for.h>
+#include <unordered_map>
+#include <vector>
+
+using namespace std;
+using namespace Eigen;
```

Comment:
Don't use the namespace, explicitly scope them. It makes the code more readable.

Reply:
```suggestion
```
If you use `std::` and `Eigen::` then you don't need these lines.